ledger/participant-integration-api/src/main/scala/platform/apiserver/StandaloneApiServer.scala

```diff
@@ -36,45 +36,118 @@ import scalaz.{-\/, \/-}
 import java.io.File
 import java.time.Clock
 
+import com.daml.ledger.participant.state.index.v2.IndexService
 import com.daml.telemetry.TelemetryContext
 
 import scala.collection.immutable
 import scala.concurrent.ExecutionContextExecutor
 import scala.util.{Failure, Success, Try}
 
-final class StandaloneApiServer(
-    ledgerId: LedgerId,
-    config: ApiServerConfig,
-    commandConfig: CommandConfiguration,
-    partyConfig: PartyConfiguration,
-    submissionConfig: SubmissionConfiguration,
-    optWriteService: Option[state.WriteService],
-    authService: AuthService,
-    healthChecks: HealthChecks,
-    metrics: Metrics,
-    timeServiceBackend: Option[TimeServiceBackend] = None,
-    otherServices: immutable.Seq[BindableService] = immutable.Seq.empty,
-    otherInterceptors: List[ServerInterceptor] = List.empty,
-    engine: Engine,
-    servicesExecutionContext: ExecutionContextExecutor,
-    lfValueTranslationCache: LfValueTranslationCache.Cache,
-    checkOverloaded: TelemetryContext => Option[state.SubmissionResult] =
-      _ => None, // Used for Canton rate-limiting
-)(implicit actorSystem: ActorSystem, materializer: Materializer, loggingContext: LoggingContext)
-    extends ResourceOwner[ApiServer] {
-
+object StandaloneApiServer {
   private val logger = ContextualizedLogger.get(this.getClass)
 
-  // Name of this participant,
-  val participantId: Ref.ParticipantId = config.participantId
-
-  override def acquire()(implicit context: ResourceContext): Resource[ApiServer] = {
-    val packageStore = loadDamlPackages()
-    preloadPackages(packageStore)
+  def apply(
+      ledgerId: LedgerId,
+      config: ApiServerConfig,
+      commandConfig: CommandConfiguration,
+      partyConfig: PartyConfiguration,
+      submissionConfig: SubmissionConfiguration,
+      optWriteService: Option[state.WriteService],
+      authService: AuthService,
+      healthChecks: HealthChecks,
+      metrics: Metrics,
+      timeServiceBackend: Option[TimeServiceBackend] = None,
+      otherServices: immutable.Seq[BindableService] = immutable.Seq.empty,
+      otherInterceptors: List[ServerInterceptor] = List.empty,
+      engine: Engine,
+      servicesExecutionContext: ExecutionContextExecutor,
+      lfValueTranslationCache: LfValueTranslationCache.Cache,
+      checkOverloaded: TelemetryContext => Option[state.SubmissionResult] =
+        _ => None, // Used for Canton rate-limiting
+  )(implicit
+      actorSystem: ActorSystem,
+      materializer: Materializer,
+      loggingContext: LoggingContext,
+  ): ResourceOwner[ApiServer] = {
+    for {
+      indexService <- this.indexService(
```

Comment:
```suggestion
      indexService <- indexService(
```

Reply:
ahh, right, because left hand side lives in a nested scope already :man_facepalming: